xgi/generators/classic.py

```diff
@@ -196,17 +206,27 @@ def flag_complex(g, max_order=2, seed=None):
     """
     # This import needs to happen when this function is called, not when it is
     # defined.  Otherwise, a circular import error would happen.
-    from ..classes import SimplicialComplex
 
-    nodes = g.nodes()
-    edges = list(g.edges())
+    nodes = G.nodes()
+    edges = G.edges()
 
     # compute all triangles to fill
-    max_cliques = list(nx.find_cliques(g))
+    max_cliques = list(nx.find_cliques(G))
 
     S = SimplicialComplex()
     S.add_nodes_from(nodes)
-    S.add_simplices_from(max_cliques, max_order=max_order)
+    S.add_simplices_from(edges)
+    if not ps: # promote all cliques
+        S.add_simplices_from(max_cliques, max_order=max_order)
+    else: # promote cliques with a given probability
+        for i, p in enumerate(ps[:max_order - 1]):
+            d = i + 2 # simplex order
+            cliques_d = [x for x in max_cliques if len(x) == d + 1]
```

Comment:
Here, you are iterating over `max_cliques` once per iteration of the loop on line 222. A better way of doing this would be to define a dictionary before the loop starts:
```python3
cliques_d = defaultdict(list)
for x in max_cliques:
    cliques_d[len(x)].append(x)
```
and then just use `cliques_d[d+1]` in each of the iterations of the loop. This should in general be faster IF we expect `max_cliques` to be large...

Reply:
Nice thanks!